libbeat/dashboards/kibana_loader.go

```diff
@@ -0,0 +1,97 @@
+package dashboards
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/setup/kibana"
+)
+
+var importAPI = "/api/kibana/dashboards/import"
+var exportAPI = "/api/kibana/dashboards/export"
+
+type KibanaLoader struct {
+	client       *kibana.Client
+	config       *DashboardsConfig
+	version      string
+	msgOutputter *MessageOutputter
+}
+
+func NewKibanaLoader(cfg *common.Config, dashboardsConfig *DashboardsConfig, msgOutputter *MessageOutputter) (*KibanaLoader, error) {
+
+	if cfg == nil {
+		cfg = common.NewConfig()
+		err := cfg.SetBool("enabled", -1, true)
+		if err != nil {
+			return nil, fmt.Errorf("Error setting setup.kibana.enabled=true: %v", err)
+		}
+	}
+
+	client, err := kibana.NewKibanaClient(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("Error creating Kibana client: %v", err)
+	}
+
+	version, err := client.GetVersion()
+	if err != nil {
+		return nil, fmt.Errorf("fail to get the Kibana version:%v", err)
+	}
+
+	loader := KibanaLoader{
+		client:       client,
+		config:       dashboardsConfig,
+		version:      version,
+		msgOutputter: msgOutputter,
+	}
+
+	loader.statusMsg("Initialize the Kibana %s loader", version)
+
+	return &loader, nil
+}
+
+func (loader KibanaLoader) ImportIndex(file string) error {
+
+	params := url.Values{}
+	params.Set("force", "true") //overwrite the existing dashboards
+
+	// read json file
+	content, err := ioutil.ReadFile(file)
+	if err != nil {
+		return fmt.Errorf("fail to read index-pattern: %v", err)
+	}
+
+	return loader.client.ImportJSON(importAPI, params, bytes.NewBuffer(content))
+
+}
+
+func (loader KibanaLoader) ImportDashboard(file string) error {
+
+	params := url.Values{}
+	params.Set("force", "true")            //overwrite the existing dashboards
+	params.Add("exclude", "index-pattern") //don't import the index pattern from the dashboards
+
+	// read json file
+	content, err := ioutil.ReadFile(file)
+	if err != nil {
+		return fmt.Errorf("fail to read index-pattern: %v", err)
+	}
+
+	return loader.client.ImportJSON(importAPI, params, bytes.NewBuffer(content))
+}
+
+func (loader KibanaLoader) Close() error {
+	//return loader.client.Close()
```

Comment:
Seems that there's nothing to do on Close. Then you can just delete the comment, not to confuse future readers.